File: abc238/b/b.py
from sys import stdin
input = stdin.readline


def read_int():
    return int(input())


def read_ints():
    return map(int, input().split())


n = read_int()
a = list(read_ints())
s = set()
s.add(0)
angle = 0
for ai in a:
    angle += ai
    angle %= 360
    s.add(angle)
v = sorted(list(s))
v.append(v[0] + 360)
print(max(y - x for x, y in zip(v[:-1], v[1:])))

File: abc238/b/b.jl
function read_ints()::Vector{Int}
    line = readline()
    return map(x -> parse(Int, x), split(line))
end

n = read_ints()[1]
angles = read_ints()

cuts = Set{Int}()
push!(cuts, 0)
now = Ref(0)
for angle in angles
    now[] = (now[] + angle) % 360
    push!(cuts, now[])
end

cuts = [sort(collect(cuts))..., 360]
println(maximum(y - x for (x, y) in zip(cuts[1:end-1], cuts[2:end])))
